pkg/controller/provider/validation.go

```diff
@@ -55,6 +55,15 @@ const (
 	SkipTLSVerification = "SkipTLSVerification"
 )
 
+//
+// Phases
+const (
+	ValidationFailed = "ValidationFailed"
+	ConnectionFailed = "ConnectionFailed"
+	Ready            = "Ready"
+	Staging          = "Staging"
+)
```

Comment:
I'm missing here 2 states that are used in the UI:
1. `pending`  - displayed as a spinner with message "Loading the inventory"
2. `running` (in the sense that there are running migrations) - used indirectly in the UI i.e. you cannot remove or edit a  provider that is being used by a running migration. This information is not available from CLI which may lead to problems. For details see https://github.com/konveyor/forklift-ui/blob/main/pkg/web/src/app/Providers/components/ProvidersTable/ProviderActionsDropdown.tsx#L40

Some questions to the provider lifecycle:
1.  does the provider need to stay connected all the time to be useful? if the connection is used to synchronize then perhaps we could rephrase the condition to reflect that we have (potentially) outdated/obsolete inventory data
2.  is it possible to have an invalid provider in the system? if not then validation is part of initialization flow (`staging`)

Reply:
`phase` is not state, it is not replacing debugging by looking up the conditions when the resource is on a "negative" flow,
it can be used to quickly filter the resources that are on a "negative" flow, and then fetch there conditions for more information.

> `pending` - displayed as a spinner with message "Loading the inventory"

`pending` can be roughly approximated by `Staging` as it indicate a transitional state that should turn to `Ready` or some fail phase, a `spinner` can be a good icon for this phase, because it should be temporary state.

> `running` (in the sense that there are running migrations) - used indirectly in the UI 

`running` in the meaning of "we currently have a running plan that use this provider", is more problematic, the provider resource does not store this information, (e.g. plans do not tag/un-tag a provider when they start/stop), I'm ok with not adding this life phase of a provider (e.g. `currently used by a running plan` phase), we may want to add in some future planning adding a feature for plans, that they will tag the provider somehow when running.
@rszwajko  we can continue the current behavior of fetching this information from the plans list, like it's done in the example you provided, in the example we do not use the provider state, we fetch all plans and then filter them to find the once that are running and using this provider.
https://github.com/konveyor/forklift-ui/blob/main/pkg/web/src/app/Providers/components/ProvidersTable/ProviderActionsDropdown.tsx#L40

Reply:
> does the provider need to stay connected all the time to be useful? if the connection is used to synchronize then perhaps > we could rephrase the condition to reflect that we have (potentially) outdated/obsolete inventory data

At this moment a provider is always connected, it is currently debated whether it's a good design feature:
pros: 
a user can know the provider is connected before starting a plan (and not just that it was connected at some point)
a UI user can have an up to date information about the cluster
cons:
lots of unneeded net traffic

(*) AFAIK, no work on that
 
> is it possible to have an invalid provider in the system? if not then validation is part of initialization flow (staging)

At this moment yes, as demonstrated in this PR description, it is currently debated whether it's a good design feature:
pros:
a user can create providers that are currently not reachable, and then connect them just when starting migration.
cons:
a user can't confirm the information (e.g. passwords, certs) are correct and may end up redoing the providers creation once the provider clusters come online.

(*) AFAIK, we do have PR, creating a web hook that will change this behavior, and will not allow creation of invalid providers.

cc:// @ahadas @michalskrivanek